nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -367,6 +367,17 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
                     'Expert parallelism is currently not supporting Apex distributed optimizer, use Mcore distributed optimizer instead'
                 )
 
+        if self.cfg.optim.get('overlap_param_gather_with_optimizer_step', False):
+            assert self.cfg.optim.get(
+                'overlap_param_sync', False
+            ), "must use overlap_param_gather_with_optimizer_step with overlap_param_sync"
+            assert (
+                self.cfg.get('virtual_pipeline_model_parallel_size', None) is not None
```

Comment:
Does it work even with `virtual_pipeline_model_parallel_size=1`?

Reply:
Tested and `virtual_pipeline_model_parallel_size=1` does not work since effectively this still not interleaved.
Followed Mocre [code](https://gitlab-master.nvidia.com/ADLR/megatron-lm/-/blob/main/megatron/training/arguments.py#L311) earlier. But you are right this need to be more strict.